backend/src/process.py

```diff
@@ -168,6 +168,52 @@ async def run(
                 "Errors occurred during iteration: \n• {}".format("\n• ".join(errors))
             )
 
+    async def run_while(
+        self,
+        length_estimate: int,
+        before: Callable[[int], Union[None, Literal[False]]],
+    ):
+        errors: List[str] = []
+        index = -1
+        while True:
+            try:
+                await self.progress.suspend()
+
+                if index < length_estimate:
+                    index += 1
+                else:
+                    length_estimate = int(length_estimate * 1.1)
```

Comment:
What's the idea here?

Also, you still skip incrementing the index sometimes, so the index of some frames will be wrong.

Reply:
Oh yeah, oops. 

The idea here is if we pass the estimate, the estimate is obviously too small so let's just increase it a bit and keep going

Reply:
Couldn't we do something like keep it at 99% until we exit the while loop? Basically, we could always do `run_iteration(min(index, length - 1), length)`.

Reply:
We could I guess, but this way shows that there's still progress going on since you can see it still increasing

Reply:
Idk which is better tbh. On the one hand, sitting at 99% is more accurate, but going backwards a bunch and then slowly going up is a bit more indicative of something actively happening.